arthur/pagination.py

```diff
@@ -0,0 +1,63 @@
+from collections.abc import Sequence
+
+import discord
+from discord.ext.commands import Context
+from pydis_core.utils.pagination import LinePaginator as _LinePaginator, PaginationEmojis
+
+from arthur.config import CONFIG
+
+
+class LinePaginator(_LinePaginator):
+    """
+    A class that aids in paginating code blocks for Discord messages.
+
+    See the super class's docs for more info.
+    """
+
+    @classmethod
+    async def paginate(
+        cls,
+        lines: list[str],
+        ctx: Context | discord.Interaction,
+        embed: discord.Embed,
+        prefix: str = "",
+        suffix: str = "",
+        max_lines: int | None = None,
+        max_size: int = 500,
+        scale_to_size: int = 4000,
+        restrict_to_user: discord.User | None = None,
+        timeout: int = 300,
+        footer_text: str | None = None,
+        url: str | None = None,
+        allowed_roles: Sequence[int] | None = None,
+        *,
+        reply: bool = False,
+        empty: bool = True,
+        exception_on_empty_embed: bool = False,
+    ) -> discord.Message | None:
```

Comment:
How come we don't make use of `*args, **kwargs` here? From my understanding we only update the emojis?

Reply:
We only update the emojis indeed, but I prefer being explicit on what's being passed down as arguments.

Reply:
My concern here is that this binds our own function's definition to `discord`s function definition, and future updates in the library may break here without us noticing.

Reply:
This signature hasn't changed in ages.

I see your point, but the main changes needs to happen in bot-core as the downstream's method signature doesn't allow for extra arguments.

We could do that in a separate PR if you'd like ?